src/backend/distributed/deparser/deparse_table_stmts.c

```diff
@@ -0,0 +1,48 @@
+/*-------------------------------------------------------------------------
+ *
+ * deparse_table_stmts.c
+ *	  All routines to deparse table statements.
+ *	  This file contains all entry points specific for table statement deparsing as well as
+ *	  functions that are currently only used for deparsing of the table statements.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres.h"
+
+#include "distributed/deparser.h"
+#include "nodes/nodes.h"
+#include "nodes/parsenodes.h"
+
+static void AppendAlterTableSchemaStmt(StringInfo buf, AlterObjectSchemaStmt *stmt);
+
+char *
+DeparseAlterTableSchemaStmt(Node *node)
+{
+	AlterObjectSchemaStmt *stmt = castNode(AlterObjectSchemaStmt, node);
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+	Assert(stmt->objectType == OBJECT_TABLE);
+
+	AppendAlterTableSchemaStmt(&str, stmt);
+	return str.data;
+}
+
+
+static void
+AppendAlterTableSchemaStmt(StringInfo buf, AlterObjectSchemaStmt *stmt)
+{
+	Assert(stmt->objectType == OBJECT_TABLE);
+	appendStringInfo(buf, "ALTER TABLE ");
+	if (stmt->missing_ok)
+	{
+		appendStringInfo(buf, "IF EXISTS ");
+	}
+	if (stmt->relation->schemaname)
+	{
+		appendStringInfo(buf, "%s.", stmt->relation->schemaname);
+	}
+	appendStringInfo(buf, "%s SET SCHEMA %s;", stmt->relation->relname, stmt->newschema);
```

Comment:
We have some potential of malformed sql here. When string formatting sql we need to take care to correctly escape user data (relation name and schema names in this case) from control data.

Here we want to use `quote_qualified_identifier` for the first part with both the old schema name and relation name, no need for `NULL` checks as this function already implements them. The second part needs to be passed through `quote_identifier` if I remember correctly.

Also good to add some tests to the test suite with characters in the schema names and table names that are outside the range where we don't have to quote. The safe characters are:
 - lower case letters
 - numbers
 - underscore.

Anything outside that range will cause issues with the current code. I think we often use double quotes in identifiers for these tests.